### ISSUE TITLE
safearray __del__ raises AttributeError on interpreter shutdown

### ISSUE THREAD

**Issue by mitaa** [enthought/comtypes#76]
```python
#a.py
from comtypes.automation import _midlSAFEARRAY
from comtypes.automation import VARIANT

foo = _midlSAFEARRAY(VARIANT).from_param([0, 1, 2])
```
```python
#b.py
class Bar:
    __metaclass__ = type
    import a
```
    Python 2.7.8 (default, Jun 30 2014, 16:03:49) [MSC v.1500 32 bit (Intel)] on win 32
    Type "help", "copyright", "credits" or "license" for more information.

    C:\test>python b.py
    Exception AttributeError: "'NoneType' object has no attribute 'SafeArrayDestroy'" in <bound method LP_SAFEARRAY_tagVARIANT.__del__ of comtypes.safearray.LP_SAFEARRAY_tagVARIANT object at 0x01FEFA30>> ignored

This seems to be the same issue as http://bugs.python.org/issue5099.

https://hg.python.org/cpython/rev/ca496bfacd1b/
>subprocess.Popen.__del__ referenced global objects, which is a no-no thanks to interpreter shutdown semantics. Same issue goes for the methods that __del__ called. Now all the methods capture the global objects it needs as default values to private parameters (could have stuck them on the class object itself, but since the objects have nothing directly to do with the class that seemed wrong). 

The same approach solves the problem for me.

```python
        def __del__(self, SafeArrayDestroy=_safearray.SafeArrayDestroy):
            if self._needsfree:
                SafeArrayDestroy(self)
```

I haven't found out yet why my example triggers this issue.

**Comment by cfarrow** [comment]
Thanks for the bug report. This is now fixed.

**Comment by cfarrow** [comment]
This is fixed by #83.
